R/function_checkers.R

```diff
@@ -0,0 +1,66 @@
+#' @title Check functions passed to exported functions
+#'
+#' @name function_checkers
+#' @rdname function_checkers
+#'
+#' @description Internal helper function that check whether a function passed as
+#' an argument meets the requirements of package methods.
+#'
+#' `test_fn_req_args()` checks whether the function has only the expected number
+#' of required arguments, i.e., arguments without default values. Defaults to
+#' checking for a single required argument.
+#'
+#' `test_fn_num_out()` checks whether a function accepting a numeric vector
+#' input returns a numeric vector output of the same length as the input, with
+#' finite non-missing values \eqn{\geq} 0.0.
+#'
+#' @param fn A function. This is expected to be a function evaluating the
+#' density of a distribution at numeric values.
+#' @param n_req_args The number of required arguments, i.e., arguments without
+#' default values.
+#' @param n The number of elements over which to evaluate the function `fn`.
+#' Defaults to 10, and `fn` is evaluated over `seq(n)`.
+#'
+#' @return A logical for whether the function `fn` fulfils conditions specified
+#' in the respective checks.
+#' @keywords internal
+test_fn_req_args <- function(fn, n_req_args = 1) {
+  # NOTE: replacing checkmate::assert_count()
```

Comment:
I think this comment can go or be reworded to say checking that the `n_req_args` is a non-negative integerish value.